File: app/src/main/java/java/android/notes/wrapper/animation/Rotation.java
package java.android.notes.wrapper.animation;

import android.view.animation.Animation;
import android.view.animation.DecelerateInterpolator;
import android.view.animation.RotateAnimation;

// пригодится ещё
public class Rotation extends RotateAnimation {
    public Rotation(float fromDegrees, float toDegrees, int pivotXType, float pivotXValue, int pivotYType, float pivotYValue) {
        super(fromDegrees, toDegrees, pivotXType, pivotXValue, pivotYType, pivotYValue);

        setDuration(36000);

        //setFillAfter(true);
        setInterpolator(new DecelerateInterpolator());
        setAnimationListener(new Animation.AnimationListener() {
            @Override
            public void onAnimationStart(Animation animation) {

            }

            @Override
            public void onAnimationEnd(Animation animation) {

            }

            @Override
            public void onAnimationRepeat(Animation animation) {

            }
        });
    }


}

File: app/src/main/java/java/android/notes/wrapper/animation/Rotation.kt
package java.android.notes.wrapper.animation

import android.view.animation.RotateAnimation
import android.view.animation.Animation
import android.view.animation.DecelerateInterpolator

// пригодится ещё
class Rotation(
    fromDegrees: Float,
    toDegrees: Float,
    pivotXType: Int,
    pivotXValue: Float,
    pivotYType: Int,
    pivotYValue: Float
) : RotateAnimation(fromDegrees, toDegrees, pivotXType, pivotXValue, pivotYType, pivotYValue) {
    init {
        duration = 36000

        //setFillAfter(true);
        interpolator = DecelerateInterpolator()
        setAnimationListener(object : AnimationListener {
            override fun onAnimationStart(animation: Animation) {}
            override fun onAnimationEnd(animation: Animation) {}
            override fun onAnimationRepeat(animation: Animation) {}
        })
    }
}